geoserver/geoserver_seeder/geoserver_seeder.rb

```diff
@@ -357,87 +49,74 @@ def seed_layers(layers, opts)
 #   - true on success, false otherwise
 #
 def geoserver_seeder_main(opts)
-  layers = []
 
-  if opts[:geonetwork]
-    # Get layers from Geonetwork URL
-    geonetwork_connector = GeonetworkConnector.new(opts[:geonetwork], opts[:geoserver])
-    layers = geonetwork_connector.wms_layers
-  else
-    # Get layers from command line
-    layers = opts[:layers]
-  end
+    geonetwork_connector = GeonetworkConnector.new(opts)
+
+    # layers that match supplied WMS server or all layers on Geonetwork
+    geonetwork_layers = geonetwork_connector.wms_layers
+    layers_to_seed = []
+
+    # add only layers from command line
+    if opts[:layers]
+        opts[:layers].each do | layer|
+            if(geonetwork_layers.any? {|geolayer| geolayer[1] == layer})
+                layers_to_seed.push([opts[:geoserver], layer])
+            else
+                $logger.error("Layer not found: #{layer} on #{opts[:geoserver]} from #{opts[:geonetwork]}")
+            end
+        end
+
+        return seed_layers(layers_to_seed, opts)
+    else
+        return seed_layers(layers, opts)
+    end
+
 
-  return seed_layers(layers, opts)
 end
 
 # Arguments parsing
 opts = Trollop::options do
-  banner <<-EOS
+    banner <<-EOS
     Seed Geoserver tiles based on results returned from Geonetwork
 
     Examples:
-        Seed Geoserver with all layers in catalogue-123.aodn.org.au, zoom level
-        2 to 5, 4 threads:
-           geoserver_seeder.rb -u https://catalogue-123.aodn.org.au/geonetwork
-             -g http://geoserver-123.aodn.org.au/geoserver/wms
-             -s 2 -t 2 -T 256 -G 20 -U 'URL_FORMAT'
+        (dry run) Seed Geowebcache with all layers in catalogue-imos.aodn.org.au, zoom level 2 to default:
+           geoserver_seeder.rb -d -u https://catalogue-imos.aodn.org.au/geonetwork -s 2 
+
+        (dry run) Seed  all layers in catalogue-imos.aodn.org.au from http://maps.aims.gov.au/geoserver/wms
+            geoserver_seeder.rb -d -u https://catalogue-imos.aodn.org.au/geonetwork -g http://maps.aims.gov.au/geoserver/wms
```

Comment:
The motivation for this work is to improve the speed of aodn contributor geoservers - when serving layers from our portal.

I guess it would be possible  to call this script multiple times for each geoserver that we intend to proxy/cache? Is that would you suggest?

Alternatively, maybe we could just default to seeding all geoservers that are listed in the metadata.